proton

```diff
@@ -457,6 +458,13 @@ def run():
         else:
             run_wine([wine_path] + sys.argv[2:])
 
+if sys.version_info[0] == 2:
```

Comment:
@nanonyme I finally bit the bullet and wrote this to mux the binary input to the correct binary output, according to your suggestion in https://github.com/ValveSoftware/Proton/pull/9#discussion_r212055974

I would have preferred to avoid explicitly different codepaths if it had peen possible.

Technically, it is possible to leave it as `sys.stdout.write(path.encode('utf-8'))`, but would cause weird issues on non utf8-encoded outputs that subprocess might spit out (or just outright fail).

Reply:
Also decoding is wrong since Python 2 expects bytes and decode results in unicode so you get implit encode inside write

Reply:
I meant `decode`, sorry. `encode` wouldn't work. But yeah, this would rely on implicit behaviour. Are you good with the changes now?

Reply:
This looks fine to me. Note that someone still needs to check ffmpeg until Proton really supports Python3

Reply:
Note you can also
from __future__ import print_function

Reply:
Mmm, good point, that could remind people to keep the code compatible... On the other hand, there are very few `print` statements. I'd rather see a CI system that checks both with python 2 and 3.

Reply:
It's just that parenthesis does different things with print statement and function which can be confusing